core/src/main/java/bisq/core/offer/PriceEventHandler.java

```diff
@@ -0,0 +1,148 @@
+    private final PriceFeedService priceFeedService;
+    private final Map<String, Set<OpenOffer>> openOffersByCurrency = new HashMap<>();
+
+    @Inject
+    public PriceEventHandler(OpenOfferManager openOfferManager, PriceFeedService priceFeedService) {
+        this.openOfferManager = openOfferManager;
+        this.priceFeedService = priceFeedService;
+    }
+
+    public void onAllServicesInitialized() {
+        openOfferManager.getObservableList().addListener((ListChangeListener<OpenOffer>) c -> {
+            c.next();
+            if (c.wasAdded()) {
+                onAddedOpenOffers(c.getAddedSubList());
+            }
+            if (c.wasRemoved()) {
+                onRemovedOpenOffers(c.getRemoved());
+            }
+        });
+        onAddedOpenOffers(openOfferManager.getObservableList());
+
+        priceFeedService.updateCounterProperty().addListener((observable, oldValue, newValue) -> onPriceFeedChanged());
+        onPriceFeedChanged();
+    }
+
+    private void onPriceFeedChanged() {
+        openOffersByCurrency.keySet().stream()
+                .map(priceFeedService::getMarketPrice)
+                .filter(Objects::nonNull)
+                .filter(marketPrice -> openOffersByCurrency.containsKey(marketPrice.getCurrencyCode()))
+                .forEach(marketPrice -> {
+                    openOffersByCurrency.get(marketPrice.getCurrencyCode()).stream()
+                            .filter(openOffer -> !openOffer.isDeactivated())
+                            .forEach(openOffer -> checkPriceThreshold(marketPrice, openOffer));
+                });
+    }
+
+    private void checkPriceThreshold(bisq.core.provider.price.MarketPrice marketPrice, OpenOffer openOffer) {
+        Price price = openOffer.getOffer().getPrice();
+        if (price == null) {
+            return;
+        }
+
+        String currencyCode = openOffer.getOffer().getCurrencyCode();
+        int smallestUnitExponent = CurrencyUtil.isCryptoCurrency(currencyCode) ?
+                Altcoin.SMALLEST_UNIT_EXPONENT :
+                Fiat.SMALLEST_UNIT_EXPONENT;
+        long marketPriceAsLong = roundDoubleToLong(
+                scaleUpByPowerOf10(marketPrice.getPrice(), smallestUnitExponent));
+        long triggerPrice = openOffer.getTriggerPrice();
+        if (triggerPrice > 0) {
+            OfferPayload.Direction direction = openOffer.getOffer().getDirection();
+            boolean triggered = direction == OfferPayload.Direction.BUY ?
+                    marketPriceAsLong > triggerPrice :
+                    marketPriceAsLong < triggerPrice;
+            if (triggered) {
+                log.error("Market price exceeded the trigger price of the open offer. " +
+                                "We deactivate the open offer with ID {}. Currency: {}; offer direction: {}; " +
+                                "Market price: {}; Upper price threshold : {}",
+                        openOffer.getOffer().getShortId(),
```

Comment:
Minor detail... placement of colons are not consistent (the last one).  
And you sure you wanted to use semi-colons to delimit key value pairs?  Is that consistent with other error log statements?

Reply:
I think we have no style rule for that...